### PR TITLE
Perform reads and modifications to Seed.Info in a concurrency safe way

### PR DESCRIPTION
**Pull request by stoyanr**
**How to categorize this PR?**

/area control-plane
/kind bug

**What this PR does / why we need it**:
* Ensures that patch operations on `Seed.Info` are always performed on a copy, and the result is written back to `Seed.Info` via a single pointer assignment. 
* Replaces usages of `o.Seed.Info` in `shoot_control.go` with the seed object that was used to initialize the operation. This is a cosmetic change that doesn't have a real functional impact (the 2 pointers are equal), but is important for consistency. 
* Replaces all naked usages of `Seed.Info` in production code that could be executed concurrently with the newly introduced methods `GetInfo()`, `UpdateInfo()`, and `UpdateInfoStatus()` that perform reading and modification in a concurrency safe way.
* Introduces another new method `SetInfo()` and replaces all naked assignments to `Seed.Info` in test code with this method.
* Renames `Info` to `info`, ensuring that it can only be read from and written to using the above methods.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
* This is a similar change for `Seed.Info` to what was introduced for `Shoot.Info` with #4459. You may want to a look at that PR as well for additional motivation, explanations, etc.
* Unlike `Shoot.Info`, there are currently no concurrent reads / writes to `Seed.Info` that might lead to issues. So the purpose of this PR is mainly to avoid falling into the same trap with `Seed.Info` in the future, if for any reason such concurrent reads and writes are introduced. It also removes the inconsistency of having `Seed.Info` but `Shoot.GetInfo()`.

**Release note**:

```other operator
Improved handling of the seed resource in the seed controller to ensure that potential data races are avoided in the future.
```

### CODE REVIEW

File: pkg/gardenlet/controller/shoot/shoot_control.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ func (c *Controller) initializeOperation(ctx context.Context, logger *logrus.Ent
 		WithShootSecretFromReader(gardenClient.Client()).
 		WithProjectName(project.Name).
 		WithExposureClassFromReader(gardenClient.Client()).
-		WithDisableDNS(!seedObj.Info.Spec.Settings.ShootDNS.Enabled).
+		WithDisableDNS(!seedObj.GetInfo().Spec.Settings.ShootDNS.Enabled).
```
Comment (timuthy) [inline pkg/gardenlet/controller/shoot/shoot_control.go]:
```suggestion
		WithDisableDNS(!seed.Spec.Settings.ShootDNS.Enabled).
```
I think both are valid but for consistency reasons in this file we can directly use `seed`.

**Review by timuthy — COMMENTED** [review]
Looks good, thank you. One nit from my side.

File: pkg/gardenlet/controller/shoot/shoot_control.go (hunk 1 above)
Comment (stoyanr) [reply pkg/gardenlet/controller/shoot/shoot_control.go]: Yes, makes sense.

Comment (stoyanr) [reply pkg/gardenlet/controller/shoot/shoot_control.go]: Done as suggested.

**Comment by stoyanr** [comment]
@timuthy I addressed your recommendation now and rebased on top of latest master. Let's merge this if there is no more feedback since it gets outdated (and therefore requires additional rebases) rather quickly.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**